lib/bloc/location/location_bloc.dart

```diff
@@ -15,21 +17,104 @@ class LocationBloc extends Bloc<LocationEvent, LocationState> {
+      await _requestPermissionLocationEvent(emit);
+    });
+    on<DefaultLocationServiceEvent>(
+        (LocationEvent event, Emitter<LocationState> emit) async {
+      await _defaultLocationServiceEvent(emit);
+    });
   }
 
   Future<void> _requestCurrentLocation(Emitter<LocationState> emit) async {
-    try {
-      final LocationDetails data = await _locationService.getCurrentLocation();
-      final String dateDisplay =
-          DateFormat('dd-MMM-yyyy').format(DateTime.now());
+    final bool isConnected = await InternetConnectionChecker().hasConnection;
+    //internet connected
+    if (isConnected) {
+      SharedPreferences prefs = await SharedPreferences.getInstance();
+      final bool isLoaded = prefs.getBool('isLoadedDialog') ?? false;
+      //not Load permission dialog
+      if (!isLoaded) {
+        try {
+          final LocationDetails data =
+              await _locationService.getCurrentLocation();
+          final String dateDisplay =
+              DateFormat('dd-MMM-yyyy').format(DateTime.now());
+          emit(LocationReadyState(data, dateDisplay));
+        } on LocationServiceDisableException catch (_) {
+          emit(LocationServiceDisableState());
+        } on LocationPermissionNotGrantedException catch (_) {
+          emit(LocationPermissionNotGrantedState());
+        } on Exception catch (_) {
+          emit(UnknownLocationErrorState());
+        }
+      }
+      //Loaded permission dialog
+      else {
+        final PermissionStatus permissionGranted =
+            await _locationService.permissionStatus();
+        final LocationDetails data =
+            await _locationService.getCurrentLocation();
+        final LocationDetails dataDefault =
+            LocationDetails(10.7840007, 106.7034988);
+        final String dateDisplay =
+            DateFormat('dd-MMM-yyyy').format(DateTime.now());
+        switch (permissionGranted) {
+          //permission granted
+          case PermissionStatus.granted:
+            emit(LocationReadyState(data, dateDisplay));
+            break;
+          //permission denied -  load default location
+          case PermissionStatus.denied:
+            emit(LocationReadyState(dataDefault, dateDisplay));
+            break;
+          //permission deniedForever -  load default location
+          case PermissionStatus.deniedForever:
+            emit(LocationReadyState(dataDefault, dateDisplay));
+            break;
+          default:
+        }
+      }
+    }
+    //internet not connected
+    else {
+      emit(InternetDisconectedState());
+    }
+  }
 
-      emit(LocationReadyState(data, dateDisplay));
-    } on LocationServiceDisableException catch (_) {
-      emit(LocationServiceDisableState());
-    } on LocationPermissionNotGrantedException catch (_) {
-      emit(LocationPermissionNotGrantedState());
-    } on Exception catch (_) {
-      emit(UnknownLocationErrorState());
+  Future<void> _requestPermissionLocationEvent(
```

Comment:
This function is good, in fact it can replace the method to request current location, because it has all logic needed to request current location.